### PR TITLE
Converting Types from js to ts

### PR DESCRIPTION
**Pull request by MegRiley**
# Summary
This PR uses existing types from [the original module builder](https://github.com/synthetichealth/module-builder/tree/master/src/types) and converted them to typescript. 
## New behavior
When running the CLI, there is no new behavior (there is no output). 
## Code changes
The following files were created: 

- attributes.ts - this file contains
- code.ts - this file defines the type for the code object that is used throughout states.ts (and with the addition of this file I updated states.ts to use this new type)
- conditional.ts- This file defines various types  such as SocioeconomicStatusConditional and more basic ones like AndConditional
- Module.ts- This file just defines the type module making use of BaseState 
- Transition.ts - This file defines the different types of transition 
- Unit.ts - this file contains the definitions for UnitOfTime and UnitOfAge types

# Testing guidance
There is no specific testing guidance for this change, however any tests that had passed previously against states.ts should still be passing

### CODE REVIEW

File: src/states/factory.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // need updated ts states file from other task
+import { code } from '../types/code';
```
Comment (mgramigna) [inline src/states/factory.ts:2]: Lint is saying this import is never used

File: src/types/code.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { type } from 'os';
```
Comment (mgramigna) [inline src/types/code.ts:1]: I think this import was added by mistake

File: src/types/conditional.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
+// @flow
```
Comment (mgramigna) [inline src/types/conditional.ts:1]: the `@flow` stuff was used for synthea-specific reasons. We don't need it 
```suggestion
```

File: src/types/code.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+export type code = {
```
Comment (mgramigna) [inline src/types/code.ts:3]:
```suggestion
export type Code = {
```

can we capitalize this type?

File: src/types/code.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  system: 'SNOMED-CT' | 'RxNorm' | 'LOINC' | 'NUBC' | 'DICOM-DCM' | 'DICOM-SOP';
+  code: string;
+  display: string;
+  value_set: string;
```
Comment (mgramigna) [inline src/types/code.ts:7]:
```suggestion
  value_set?: string;
```

I think value_set should be optional

**Review by mgramigna — CHANGES_REQUESTED** [review]
small comments

**Review by hossenlopp — APPROVED** [review]

**Pull request merged**